crypto/dh/dh_key.c

```diff
@@ -369,8 +386,23 @@ static int generate_key(DH *dh)
     if (!ossl_dh_generate_public_key(ctx, dh, priv_key, pub_key))
         goto err;
 
+#ifdef FIPS_MODULE
+    if (DH_check_pub_key(dh, pub_key, &validate) <= 0) {
+        ERR_raise(ERR_LIB_DH, DH_R_CHECK_PUBKEY_INVALID);
+        goto err;
+    }
+#endif
+
     dh->pub_key = pub_key;
     dh->priv_key = priv_key;
+#ifdef FIPS_MODULE
+    if (ossl_dh_check_pairwise(dh) <= 0) {
+        ossl_set_error_state(OSSL_SELF_TEST_TYPE_PCT);
+        ERR_raise(ERR_LIB_PROV, DH_R_CHECK_PUBKEY_INVALID);
+        goto err;
+    }
+#endif
+
     dh->dirty_cnt++;
```

Comment:
Why both pairwise and pub key check would be needed? That does not make sense at all to me.